tests/pruners_tests/test_median.py

```diff
@@ -120,16 +122,18 @@ def test_median_pruner_interval_steps(
     study = optuna.study.create_study()
 
     trial = optuna.trial.Trial(study, study._storage.create_new_trial(study._study_id))
```

Comment:
If I understand correctly, we can use `optuna.trial.create_trial` here, and I think we should use it instead as per the disclaimer of https://optuna.readthedocs.io/en/stable/reference/generated/optuna.trial.Trial.html#optuna.trial.Trial (quote follows).

> Note that the direct use of this constructor is not recommended. This object is seamlessly instantiated and passed to the objective function behind the optuna.study.Study.optimize() method; hence library users do not care about instantiation of this object.

This opinion is based on my take that we should not use some API if it is discouraged or deprecated.
So, what's your take?

Reply:
I do not think we can replace the code by `optuna.trial.create_trial`, because it will create `FrozenTrial`, not `Trial`. Also, the above note is for library users, not for developers. So, imo, we can use `optuna.trial.Trial` here for unittest.

Reply:
If we want to replace the code by something, it would be `trial = study._ask()`, although I do not think we should do it.

Reply:
`study.ask()` (introduced by #2158) is better, but I think replacement is optional.

Reply:
Ok. I use `study.ask()` instead of using constructor.